asmcomp/arm64/emit.mlp

```diff
@@ -114,7 +114,8 @@ let emit_addressing addr r =
   | Iindexed ofs ->
       `[{emit_reg r}, #{emit_int ofs}]`
   | Ibased(s, ofs) ->
-      `[{emit_reg r}, #:lo12:{emit_symbol_offset s ofs}]`
+      if !Clflags.dlcode then `[{emit_reg r}, #{emit_int ofs}]`
```

Comment:
This will be rejected by the assembler if `ofs` is too large.  However, I think you can just `assert (not !Clflags.dlcode)` there, see my general comment on this PR.